lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -40,33 +40,53 @@ sub new {
     my $self = {};
     bless( $self, $type );
 
-    my $config = Zonemaster::Backend::Config->load_config();
-
-    if ( $params && $params->{db} ) {
-        eval {
-            eval "require $params->{db}";
-            die "$@ \n" if $@;
-            $self->{db} = "$params->{db}"->new( { config => $config } );
-        };
-        if ($@) {
-            handle_exception('new', "Failed to initialize the [$params->{db}] database backend module: [$@]", '001');
+    if ( $params ) {
+        if ( ! $params->{config} ) {
+            handle_exception('new', "Missing 'config' parameter", '000');
         }
-    }
-    else {
-        eval {
-            my $backend_module = "Zonemaster::Backend::DB::" . $config->BackendDBType();
-            eval "require $backend_module";
-            die "$@ \n" if $@;
-            $self->{db} = $backend_module->new( { config => $config } );
-        };
-        if ($@) {
-            handle_exception('new', "Failed to initialize the [$params->{db}] database backend module: [$@]", '002');
+        $self->{config} = $params->{config};
+        if ( $params->{dbtype} ) {
+            $self->{dbtype} = _check_db($params->{dbtype});
         }
+        $self->_init_db;
     }
 
     return ( $self );
 }
 
+sub _check_db {
+    my $dbtype = shift;
+
+    if (lc $dbtype eq "sqlite" ) {
+        return "SQLite";
+    }
+    elsif (lc $dbtype eq "mysql" ) {
+        return "MySQL";
+    }
+    elsif (lc $dbtype eq "postgresql" ) {
+        return "PostgreSQL";
+    }
+    else {
+        handle_exception('_check_db', "Unknown dbtype '$dbtype', should be one of SQLite, MySQL or PostgreSQL", '001');
+    }
+}
+
+sub _init_db {
+    my $self = shift;
+    my $config = $self->{config};
+    $self->{dbtype} //= _check_db($config->BackendDBType());
```

Comment:
Eventually I'd like to put the responsibility to perform this validation on the Config module. It's okay to have it here for now though.

Reply:
I agree, this is not the place of such a test. I might move it the the Config module if I implement the same process in TestAgent.pm.